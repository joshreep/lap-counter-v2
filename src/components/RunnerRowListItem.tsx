import { RunnerRow } from '@/database/types'
import { Cell, Row } from './RunnerRowListItem.style'
import { FC } from 'react'
import Link from 'next/link'

type RunnerRowListItemProps = {
  item: RunnerRow
}

const RunnerRowListItem: FC<RunnerRowListItemProps> = ({ item }) => {
  return (
    <Link
      href={{
        pathname: '/edit-modal',
        query: { runnerId: item.runnerId, name: item.name ?? '', lapCount: item.lapCount },
      }}
    >
      <Row>
        <Cell>{item.runnerId}</Cell>
        <Cell $greedy>{item.name}</Cell>
        <Cell>
          {item.lapCount} lap{item.lapCount !== 1 && 's'}
        </Cell>
      </Row>
    </Link>
  )
}

export default RunnerRowListItem
