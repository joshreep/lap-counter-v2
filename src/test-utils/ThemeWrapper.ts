import MatchMediaMock from 'jest-matchmedia-mock'

type Theme = 'light' | 'dark'

const themes: Theme[] = ['light', 'dark']

export function forEachTheme(callback: (theme: Theme) => void) {
  themes.forEach((theme) => {
    const matchMedia = new MatchMediaMock()
    matchMedia.useMediaQuery(`(prefers-color-scheme: ${theme})`)

    callback(theme)

    matchMedia.clear()
  })
}
