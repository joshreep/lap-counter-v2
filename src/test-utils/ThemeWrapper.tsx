import SCDarkTheme from '@/theme/dark-theme'
import SCLightTheme from '@/theme/light-theme'
import {
  DarkTheme as RNDarkTheme,
  DefaultTheme as RNDefaultTheme,
  ThemeProvider as RNThemeProvider,
} from '@react-navigation/native'
import { FC, PropsWithChildren } from 'react'
import { ColorSchemeName } from 'react-native'
import { ThemeProvider as SCThemeProvider } from 'styled-components/native'

export const themes: ColorSchemeName[] = ['light', 'dark']

export function getThemeWrapper(theme: ColorSchemeName) {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => {
    return (
      <RNThemeProvider value={theme === 'dark' ? RNDarkTheme : RNDefaultTheme}>
        <SCThemeProvider theme={theme === 'dark' ? SCDarkTheme : SCLightTheme}>{children}</SCThemeProvider>
      </RNThemeProvider>
    )
  }

  return Wrapper
}

type ForEachThemeCallback = (theme: ColorSchemeName, wrapper: FC<PropsWithChildren>) => void

export function forEachTheme(callback: ForEachThemeCallback) {
  themes.forEach((theme) => {
    callback(theme, getThemeWrapper(theme))
  })
}
