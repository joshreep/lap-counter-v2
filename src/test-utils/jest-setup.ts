import '@testing-library/react-native/extend-expect'
import '@testing-library/jest-native/extend-expect'

jest.mock('react-native-reanimated', () => null, { virtual: true })
jest.mock('@testing-library/jest-native/extend-expect', () => null, { virtual: true })
