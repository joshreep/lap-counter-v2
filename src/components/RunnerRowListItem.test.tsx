import { forEachTheme } from '@/test-utils/ThemeWrapper'
import { render } from '@testing-library/react'
import RunnerRowListItem from './RunnerRowListItem'
import { ReactElement } from 'react'

forEachTheme((theme, wrapper) => {
  describe(`${theme} theme`, () => {
    test('should match snapshot', () => {
      const container = render(
        RunnerRowListItem({
          index: 0,
          item: { runnerId: '1', name: 'George Washington', lapCount: 100 },
          separators: { highlight() {}, unhighlight() {}, updateProps() {} },
        }) as ReactElement,
        { wrapper },
      )
      expect(container.toJSON()).toMatchSnapshot()
    })

    test('should render with no name prop', () => {
      const container = render(
        RunnerRowListItem({
          index: 0,
          item: { runnerId: '1', lapCount: 100 },
          separators: { highlight() {}, unhighlight() {}, updateProps() {} },
        }) as ReactElement,
        { wrapper },
      )
      expect(container.toJSON()).toMatchSnapshot()
    })
  })
})
