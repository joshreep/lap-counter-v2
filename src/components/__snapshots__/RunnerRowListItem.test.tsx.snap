// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark theme should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  href="/edit-modal?runnerId=1&name=George%20Washington&lapCount=100"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="link"
  style={
    {
      "backgroundColor": "rgba(255, 255, 255, .07)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "rgb(255, 255, 255)",
        "flexGrow": 0,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    1
  </Text>
  <Text
    greedy={true}
    style={
      {
        "color": "rgb(255, 255, 255)",
        "flexGrow": 2,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    George Washington
  </Text>
  <Text
    style={
      {
        "color": "rgb(255, 255, 255)",
        "flexGrow": 0,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    100
     lap
    s
  </Text>
</View>
`;

exports[`dark theme should render with no name prop 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  href="/edit-modal?runnerId=1&name=&lapCount=100"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="link"
  style={
    {
      "backgroundColor": "rgba(255, 255, 255, .07)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "rgb(255, 255, 255)",
        "flexGrow": 0,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    1
  </Text>
  <Text
    greedy={true}
    style={
      {
        "color": "rgb(255, 255, 255)",
        "flexGrow": 2,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  />
  <Text
    style={
      {
        "color": "rgb(255, 255, 255)",
        "flexGrow": 0,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    100
     lap
    s
  </Text>
</View>
`;

exports[`light theme should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  href="/edit-modal?runnerId=1&name=George%20Washington&lapCount=100"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="link"
  style={
    {
      "backgroundColor": "rgba(0, 0, 0, .05)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#111",
        "flexGrow": 0,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    1
  </Text>
  <Text
    greedy={true}
    style={
      {
        "color": "#111",
        "flexGrow": 2,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    George Washington
  </Text>
  <Text
    style={
      {
        "color": "#111",
        "flexGrow": 0,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    100
     lap
    s
  </Text>
</View>
`;

exports[`light theme should render with no name prop 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  href="/edit-modal?runnerId=1&name=&lapCount=100"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  role="link"
  style={
    {
      "backgroundColor": "rgba(0, 0, 0, .05)",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#111",
        "flexGrow": 0,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    1
  </Text>
  <Text
    greedy={true}
    style={
      {
        "color": "#111",
        "flexGrow": 2,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  />
  <Text
    style={
      {
        "color": "#111",
        "flexGrow": 0,
        "fontSize": 16,
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
      }
    }
  >
    100
     lap
    s
  </Text>
</View>
`;
